### DISCUSSION TITLE
Log  management on a Manager node

### DISCUSSION THREAD

**Discussion by kiaraho** [Security-Onion-Solutions/securityonion#3997] [Unsupported Versions] [answered]
Since `so-curator` does not run on the Manager node in a distributed environment, what actually keeps the Elasticsearch `log_size_limit` in check? I recently noticed the storage was getting full on a Manager node with a 500GB disk and  `log_size_limit: 325`.

For now I manually delete old indices via Kibana to prevent the disk from filling up. Is there a script that runs on Manager nodes that can do the same?

**Comment by weslambert** [comment]
`so-curator` runs on the search nodes, unless you perform an install w/ a traditional cluster, in which we intend on only running curator on the manager.

With regard to the manager filling up, what indices might be filling ES? Most of the data should be going to search nodes.  Otherwise, you may want to do the following to get a better idea:

`sudo -i`
`du -hs /* | sort -h`

**Reply by kiaraho** [reply]
It was my understanding too that most of the data should be going to the Search node in a distributed environment. I noticed when I delete older `so-ossec-*` and `so-zeek-*` indices via Kibana, the storage on the Manager frees up considerably, but nothing much happens on the Search node. The Search node is not the problem though, as `so-curator` seems to have kept things in check there.

Here is output from running the `du -hs /* | sort -h` command on the Manager.

```
0       /initrd.img
0       /initrd.img.old
0       /proc
0       /sys
0       /vmlinuz
0       /vmlinuz.old
4.0K    /lib64
4.0K    /media
4.0K    /mnt
4.0K    /srv
16K     /lost+found
56K     /tmp
84K     /dev
2.4M    /run
6.4M    /etc
15M     /bin
18M     /sbin
20M     /home
140M    /boot
314M    /lib
415M    /root
1.3G    /snap
2.4G    /usr
13G     /opt
60G     /var
389G    /nsm
```

**Reply by kiaraho** [reply]
@weslambert, I ran the `du -h -d1 /nsm/ | sort -h` command and it looks like Elasticsearch is what is using up most of the storage on the Manager node, despite me configuring `log_size_limit: 300` in its minion file. What could be causing Elasticsearch to ignore this limit?
```
4.0K    /nsm/import
4.0K    /nsm/pcap
4.0K    /nsm/pcapout
20K     /nsm/logstash
388K    /nsm/osquery
2.3M    /nsm/grafana
7.3M    /nsm/thehive
245M    /nsm/mysql
822M    /nsm/backup
1.1G    /nsm/wazuh
2.6G    /nsm/influxdb
4.1G    /nsm/soc
23G     /nsm/docker-registry
358G    /nsm/elasticsearch
389G    /nsm/
```

**Comment by weslambert** [comment]
If Curator isn't actually deleting like it should, I would expect that the issue is with the closed delete script.  We put out a hotfix for this here? Have you already updated or applied this hotfix?

https://github.com/Security-Onion-Solutions/securityonion/discussions/3738

**Reply by kiaraho** [reply]
We are still running `2.3.40` and have not applied this hotfix. However, before we upgrade to `2.3.50`, I wanted to clarify if `so-curator` actually runs on the Manager node in a distributed environment. `so-status` shows it running on the Search node, but not the Manager node.

**Reply by kiaraho** [reply]
We upgraded to `2.3.50` a couple of days ago. I can confirm the `/nsm/elasticsearch` directory is still growing uncontrollably on the Manger node, so we have to manually monitor and delete old `so-ossec` indices to keep the disk from hitting the 98% flood_stage limit.

Strangely, even after deleting a specific `so-ossec-YYYY.MM.DD` index, we are still able to access the host logs from that date in Kibana. We suspect it's because the logs are still present on the Search node until Curator removes them. This would imply the Manager node is acting as ManagerSearch node and storing a duplicate of the logs that were destined for the Search node.

@weslambert, could forwarding logs to an external destination like we did in #3456 have triggered a bug where the Manager node stores the cloned logs locally?

**Reply by kiaraho** [reply]
@weslambert, it seems #3738 only affects Standalone installations or a combined Manager/Search node. We are running a dedicated Manager with a separate Search node, yet the `/nsm/elasticsearch` folder on the manager is growing uncontrollably. The Search node has no issue self-regulating its storage via Curator. Is there anything else can we do to address this?

**Comment by wtl06639** [comment]
I am running 2.3.50 as well an am experiencing the same issue. Any fixes for this besides manually removing ?

**Reply by kiaraho** [reply]
Given this has become a critical issue for us, we wrote the bash script below to delete old OSSEC indices on the Manager, since they take up the most disk space. A cronjob then runs this script every few days.
```
ossec_index=`curl -s -k -H "Accept: application/json" https://localhost:9200/_cat/indices | jq -r . | grep so-ossec- | cut -d'"' -f4`
for i in ${ossec_index[@]}; do
  echo "Deleting ${i} index..."
  curl -s -k -X DELETE https://localhost:9200/$i
  echo -e "\nDone.\n"
done
```

**Comment by kiaraho** [comment, marked as answer]
Not sure what changed in the latest `2.3.100` release, but we noticed `/nsm` disk usage no longer exceeds ~80% on the Manager. It is now automatically pruning old logs without the need for manual intervention.
